### DISCUSSION TITLE
SSL isn't working

### DISCUSSION THREAD

**Discussion by ghost** [requarks/wiki#3922] [Error / Bug Report]
It's very confusing how SSL works with WikiJS.

There is a SSL setting in the Admin panel sadly there is no cert or key upload nor is it saying if SSL is enabled or disabled.

I know how to make Let's Encrypt SSL work on Nginx but I have no idea how SSL works with WikiJS.

**Comment by vwlau** [comment]
Going to preface this by saying I'm no expert. I just kept googling solutions and trying things until I found something that works. I installed WikiJS via docker on a Debian 10 server.

WIki.js can be accessed directly or behind a reverse proxy. 

When accessed directly, I started my docker container using this command (that I found [here](https://docs.requarks.io/install/ubuntu)). Note: I also followed this guide to set up the PostgreSQL database. You will need to pass different env variables in your command if you did something different. 

`docker create --name=wiki -e LETSENCRYPT_DOMAIN=wiki.example.com -e LETSENCRYPT_EMAIL=admin@example.com -e SSL_ACTIVE=1 -e DB_TYPE=postgres -e DB_HOST=db -e DB_PORT=5432 -e DB_PASS_FILE=/etc/wiki/.db-secret -v /etc/wiki/.db-secret:/etc/wiki/.db-secret:ro -e DB_USER=wiki -e DB_NAME=wiki -e UPGRADE_COMPANION=1 --restart=unless-stopped -h wiki --network=wikinet -p 3000:3000 requarks/wiki:2`

If you scroll down on the guide I linked earlier a bit, you will see that you can enable HTTP to HTTPS redirection on the SSL pane of the admin area like you found. 

When using a reverse proxy, you let NGINX handle the SSL, not WikiJS. First, I set up the a basic reverse proxy in the nginx.conf file. 

```
server {
    listen 80;
    server_name wiki.example.com;

    location / {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_pass http://127.0.0.1:3000;
    }   
}

```
Then I followed the instructions [here](https://certbot.eff.org/lets-encrypt/debianbuster-nginx) to get my SSL certificates and automatically reconfigure Nginx to redirect from HTTP to HTTPS. After that I got the lock symbol when I go to my wiki. You don't need to do anything in the SSL pane of the admin area in this case. I think. It will show that only the HTTP port is open but that's okay because as I understand it, it is interfacing with only the Nginx reverse proxy not the outside world.

Following these instructions, I got an A rating on https://www.ssllabs.com/ssltest/

I hope that was of some help.
